### PR TITLE
Fix appointment booking loading

### LINKED ISSUE #5409: Fix Loading animation of appointment booking
### Is your feature request related to a problem? Please describe.

When booking an appointment, After filling the information and clicking the 'Book the appointment'  button.
The button stays enabled (You can start multiple requests ) and no indication that it's loading.
image

### Describe the solution you'd like

Disable the button and add loading animation 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hamza221**
closes #5409

### CODE REVIEW

File: src/components/Appointments/AppointmentDetails.vue [hunk 1]
```diff
@@ -131,21 +134,15 @@ export default {
 		},
 	},
 	methods: {
-		async save() {
-			this.canClose = false
-			this.isLoading = true
-			try {
-				await this.$emit('save', {
```
Comment (ChristophWurst) [inline src/components/Appointments/AppointmentDetails.vue:138]: so the issue was that event listeners are not awaitable?

**Review by ChristophWurst — APPROVED** [review]
:+1:

**Comment by ChristophWurst** [comment]
cc @pjrobertson for testing and review :)

**Comment by pjrobertson** [comment]
Looking into it :)

Still struggling to get a proper dev environment set up though. Is https://github.com/juliushaertl/nextcloud-docker-dev the best way to set up an environment?

**Comment by hamza221** [comment]
> Looking into it :)
> 
> Still struggling to get a proper dev environment set up though. Is https://github.com/juliushaertl/nextcloud-docker-dev the best way to set up an environment?

That's what I'm using and I can totally recommend it

File: src/components/Appointments/AppointmentDetails.vue (hunk 1 above)
Comment (hamza221) [reply src/components/Appointments/AppointmentDetails.vue:138]: yes

**Comment by pjrobertson** [comment]
LGTM @ChristophWurst, I think it's good to merge!

Also apologies, my PR didn't get it working correctly.

**Comment by hamza221** [comment]
@pjrobertson  Thank you for the review

**Pull request merged**